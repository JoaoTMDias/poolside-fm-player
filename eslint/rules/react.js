/**
 * React specific rules
 *
 * @author Jo√£o Dias (joao.dias@wit-software.com)
 * @since 1.0.0
 */
module.exports = {
	extends: ['react-app', 'airbnb'],
	plugins: ['react', 'react-hooks'],
	rules: {
		'react/jsx-wrap-multilines': 0,
		'react/jsx-closing-bracket-location': 0,
		'react/jsx-curly-spacing': 0,
		'react/jsx-equals-spacing': 0,
		'react/jsx-first-prop-new-line': 0,
		'react/jsx-indent': 0,
		'react/jsx-indent-props': 0,
		'react/jsx-max-props-per-line': 0,
		'react/jsx-tag-spacing': 0,
		'react/jsx-closing-tag-location': 0,
		'react/jsx-filename-extension': [
			'error',
			{
				extensions: ['.js', '.jsx', '.ts', '.tsx'],
			},
		],
		'react-hooks/rules-of-hooks': 'error',
		'react-hooks/exhaustive-deps': 'warn',
		'react/prop-types': 'off',
		'react/require-default-props': 'off',

		'jsx-a11y/href-no-hash': 'off',
		'jsx-a11y/anchor-is-valid': [
			'warn',
			{
				aspects: ['invalidHref'],
			},
		],
	},
};
