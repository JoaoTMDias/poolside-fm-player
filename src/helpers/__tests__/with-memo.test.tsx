import * as React from "react";
import { shallow } from "enzyme";

import { withMemo } from "../index";

interface IDemoComponentProps {
	title: string;
	value?: string;
}

const DemoComponent: React.FunctionComponent<IDemoComponentProps> = ({ title, value }) => {
	return (
		<fieldset id="demo-component">
			<input
				id="demo-component-input"
				data-testid="demo-component-input"
				className="demo-component-input"
				type="text"
				value={value}
				placeholder="Foobar"
			/>
			<button type="button" id="demo-component-button" data-testid="demo-component-button">
				<span className="demo-component-span">{title}</span>
			</button>
		</fieldset>
	);
};

DemoComponent.defaultProps = {
	title: "Default title",
	value: "Default value",
};

const MemoDemo = withMemo<IDemoComponentProps>(DemoComponent, ["title"]);

describe("withMemo", () => {
	it("should return a MemoExoticComponent", () => {
		const component = shallow(<MemoDemo title="A new title" />);
		expect(component).toMatchSnapshot();
	});
});
