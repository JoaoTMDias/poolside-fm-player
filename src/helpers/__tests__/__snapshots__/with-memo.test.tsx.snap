// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withMemo should return a MemoExoticComponent 1`] = `ShallowWrapper {}`;
